File: 342-power-of-four/power-of-four.py
class Solution:
    def isPowerOfFour(self, n: int) -> bool:
        if n == 1:
            return True
        if n%2 != 0:
            return False
        if n>1 and n%2 == 0:
            return self.isPowerOfFour(n/4)
        return False
        
        

File: 342-power-of-four/power-of-four.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfFour = function(n) {
    if (n == 1)
    return true;
    if (n%2 != 0)
    return false;
    if (n>1 && n%2 == 0)
    return isPowerOfFour(n/4);
    return false
};